File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/repositories/PhotoDataRepository.java
package nl.acidcats.tumblrlikes.core.repositories;

import android.support.annotation.Nullable;

import java.util.List;

import nl.acidcats.tumblrlikes.core.constants.FilterType;
import nl.acidcats.tumblrlikes.core.models.Photo;
import rx.Observable;

/**
 * Created by stephan on 11/04/2017.
 */

public interface PhotoDataRepository {
    boolean hasPhoto(long postId);

    List<Photo> storePhotos(List<Photo> photos);

    List<Photo> getAllPhotos();

    long getPhotoCount();

    Photo getNextPhoto();

    void setPhotoViewStartTime(long id, long currentTime);

    void updatePhotoViewTime(long id, long currentTime);

    void setPhotoLiked(long id, boolean isLiked);

    void setPhotoFavorite(long id, boolean isFavorite);

    void hidePhoto(long id);

    @Nullable
    Photo getPhotoById(long id);

    void setFilterType(FilterType filterType);

    FilterType getFilterType();

    boolean hasUncachedPhotos();

    Photo getNextUncachedPhoto();

    void markAsCached(long id, String path);

    Observable<Void> removeCachedHiddenPhotos();

    boolean isPhotoCacheMissing(Photo photo);

    List<Long> setPhotosUncached(final List<Long> idList);

    List<Photo> getCachedPhotos();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/repositories/PhotoDataRepository.kt
package nl.acidcats.tumblrlikes.core.repositories

import nl.acidcats.tumblrlikes.core.constants.FilterType
import nl.acidcats.tumblrlikes.core.models.Photo
import rx.Observable

/**
 * Created on 24/10/2018.
 */
interface PhotoDataRepository {
    fun hasPhoto(postId: Long): Boolean

    fun storePhotos(photos: List<Photo>): List<Photo>

    fun getAllPhotos(): List<Photo>

    fun getPhotoCount(): Long

    fun getNextPhoto(): Photo?

    fun setPhotoViewStartTime(id: Long, currentTime: Long)

    fun updatePhotoViewTime(id: Long, currentTime: Long)

    fun setPhotoLiked(id: Long, isLiked: Boolean)

    fun setPhotoFavorite(id: Long, isFavorite: Boolean)

    fun hidePhoto(id: Long)

    fun getPhotoById(id: Long): Photo?

    fun setFilterType(filterType: FilterType)

    fun getFilterType(): FilterType

    fun hasUncachedPhotos(): Boolean

    fun getNextUncachedPhoto(): Photo?

    fun markAsCached(id: Long, path: String)

    fun removeCachedHiddenPhotos(): Observable<Boolean>

    fun isPhotoCacheMissing(photo: Photo): Boolean

    fun setPhotosUncached(idList: List<Long>): List<Long>

    fun getCachedPhotos(): List<Photo>
}